webapp/src/views/projects/translations/comments/Comments.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { useQueryClient } from 'react-query';
+import { container } from 'tsyringe';
+import { T } from '@tolgee/react';
+import { makeStyles, TextField, IconButton } from '@material-ui/core';
+import { Send } from '@material-ui/icons';
+
+import { parseErrorResponse } from 'tg.fixtures/errorFIxtures';
+import { useProject } from 'tg.hooks/useProject';
+import {
+  useApiInfiniteQuery,
+  useApiMutation,
+} from 'tg.service/http/useQueryApi';
+import { components } from 'tg.service/apiSchema.generated';
+import { MessageService } from 'tg.service/MessageService';
+import LoadingButton from 'tg.component/common/form/LoadingButton';
+import { SmoothProgress } from 'tg.component/SmoothProgress';
+import { useUser } from 'tg.hooks/useUser';
+import { useProjectPermissions } from 'tg.hooks/useProjectPermissions';
+import { ProjectPermissionType } from 'tg.service/response.types';
+import { useTranslationsDispatch } from '../context/TranslationsContext';
+import { Comment } from './Comment';
+
+type TranslationCommentModel = components['schemas']['TranslationCommentModel'];
+type TranslationViewModel = components['schemas']['TranslationViewModel'];
+type LanguageModel = components['schemas']['LanguageModel'];
+
+const messaging = container.resolve(MessageService);
+
+const useStyles = makeStyles((theme) => {
+  const borderColor = theme.palette.grey[200];
+  return {
+    container: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexGrow: 1,
+      overflow: 'hidden',
+    },
+    reverseScroller: {
+      display: 'flex',
+      flexDirection: 'column-reverse',
+      overflowY: 'auto',
+      overflowX: 'hidden',
+      flexBasis: 100,
+      flexGrow: 1,
+      overscrollBehavior: 'contain',
+      // scrollBehavior: 'smooth',
+    },
+    loadMore: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'flex-end',
+      minHeight: 50,
+    },
+    bottomPanel: {
+      padding: '3px 5px 3px 12px',
```

Comment:
We should use global theming constants for padding and margin `theme.spacing()`. If it doesn't satisfy our needs, maybe we should consider different  value for the global spacing unit, which is now 8px.

Reply:
It ain't that simple.

Reply:
We need quite detailed spacing e.g. here: https://tolgee.myjetbrains.com/youtrack/issue/TGX-16

Reply:
Maybe we could set it to 4. But I'm afraid that it would require a lot of changes.

Reply:
In this specific case, it's file to fine-tune it. I just would like to be sure, we are following material-ui standards to avoid future problems.

Reply:
Ok